docs/src/extend/plugins.md

```diff
@@ -8,180 +8,249 @@ eleventyNavigation:
+Plugins can expose [processors](custom-processors) for use in configuration file by providing a `processors` object. Similar to rules, each key in the `processors` object is the name of a processor and each value is the processor object itself. Here's an example:
 
 ```js
-// eslint-plugin-myPlugin
-
-module.exports = {
-    configs: {
-        myConfig: {
-            plugins: ["myPlugin"],
-            env: ["browser"],
-            rules: {
-                semi: "error",
-                "myPlugin/my-rule": "error",
-                "eslint-plugin-myPlugin/another-rule": "error"
-            }
-        },
-        myOtherConfig: {
-            plugins: ["myPlugin"],
-            env: ["node"],
-            rules: {
-                "myPlugin/my-rule": "off",
-                "eslint-plugin-myPlugin/another-rule": "off",
-                "eslint-plugin-myPlugin/yet-another-rule": "error"
-            }
-        }
+const plugin = {
+    meta: {
+        name: "eslint-plugin-example",
+        version: "1.2.3"
     },
-    rules: {
-        "my-rule": {/* rule definition */},
-        "another-rule": {/* rule definition */},
-        "yet-another-rule": {/* rule definition */}
+    processors: {
+        "processor-name": {
+            preprocess(text, filename) {/* ... */},
+            postprocess(messages, filename) { /* ... */ },
+        }
     }
 };
-```
 
-Plugins cannot force a specific configuration to be used. Users must manually include a plugin's configurations in their configuration file.
+// for ESM
+export default plugin;
 
-If the example plugin above were called `eslint-plugin-myPlugin`, the `myConfig` and `myOtherConfig` configurations would then be usable in a configuration file by extending `"plugin:myPlugin/myConfig"` and `"plugin:myPlugin/myOtherConfig"`, respectively.
+// OR for CommonJS
+module.exports = plugin;
+```
 
-```json
-// .eslintrc.json
+In order to use a processor from a plugin in a configuration file, import the plugin and include it in the `plugins` key, specifying a namespace. Then, use that namespace to reference the processor in the `processor` configuration, like this:
 
-{
-    "extends": ["plugin:myPlugin/myConfig"]
-}
+```js
+// eslint.config.js
+import example from "eslint-plugin-example";
 
+export default [
+    {
+        plugins: {
+            example
+        },
+        processor: "example/processor-name"
+    }
+];
 ```
 
-### Meta Data in Plugins
+### Configs in Plugins
 
-For easier debugging and more effective caching of plugins, it's recommended to provide a name and version in a `meta` object at the root of your plugin, like this:
+You can bundle configurations inside a plugin by specifying them under the `configs` key. This can be useful when you want to bundle a set of custom rules with a configuration that enables the recommended options. Multiple configurations are supported per plugin.
```

Comment:
```suggestion
You can bundle configurations inside a plugin by specifying them under the `configs` key. This can be useful when you want to bundle a set of custom rules with a configuration that enables the recommended rules. Multiple configurations are supported per plugin.
```

Did you mean recommended rules? Recommended rule options are usually the default for the rule.

Reply:
I was using "options" generically because you configs can specify all kinds of options and not just rules.